### PR TITLE
migrated to mvp

### PR DESCRIPTION
**Pull request by alanhala**
### Summary
- Removed logic from Fragments and added a Presenter for each View.

### Trello Card
https://trello.com/c/6bxvsVj1/29-n-13-migracion-a-patron-mvp

### CODE REVIEW

File: WNews/app/src/main/java/ar/com/wolox/wnews/activities/RouteActivity.java [hunk 1]
```diff
@@ -13,6 +15,10 @@
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         Intent intent;
+        SharedPreferences mySPrefs = PreferenceManager.getDefaultSharedPreferences(this);
```
Comment (juanignaciomolina) [inline WNews/app/src/main/java/ar/com/wolox/wnews/activities/RouteActivity.java]: Why are you doing this? Wouldn't this force `userLogged()` to always return false?

**Comment by juanignaciomolina** [comment]
There are some code style errors in this PR, remember to use the provided Java linter before creating a new PR.

File: WNews/app/src/main/java/ar/com/wolox/wnews/views/SignupView.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+    void userAlreadyExistsError();
+
+    void requiredEmailError();
+
+    void requiredPasswordError();
+
+    void requiredPasswordConfirmationError();
+
+    void showInvalidEmailError();
```
Comment (juanignaciomolina) [inline WNews/app/src/main/java/ar/com/wolox/wnews/views/SignupView.java]: It's preferable to have methods names that doesn't tell the `View` what it should do. Remember that the `Presenter` should know as little as possible about how the `View` is displaying the UI. Maybe the view doesn't even want to display an error, maybe it just logs the error on the console and the user is not aware of this.

I think that a better method name would be `onInvalidEmail()`, and the same applies to other method names, like `showDifferentPasswordsError`. What do you think?

File: WNews/app/src/main/java/ar/com/wolox/wnews/activities/RouteActivity.java (hunk 1 above)
Comment (alanhala) [inline WNews/app/src/main/java/ar/com/wolox/wnews/activities/RouteActivity.java]: @juanignaciomolina My bad. I forgot to remove that

File: WNews/app/src/main/java/ar/com/wolox/wnews/views/SignupView.java (hunk 2 above)
Comment (alanhala) [inline WNews/app/src/main/java/ar/com/wolox/wnews/views/SignupView.java]: ok, I like it

**Comment by alanhala** [comment]
@juanignaciomolina I run linter, but it didn't find any style errors

**Comment by juanignaciomolina** [comment]
@alanhala 👍

**Comment by emalamela** [comment]
Vamos alancito!

**Pull request merged**